subprojects/gradle/runner/device-provider/src/main/kotlin/com/avito/android/runner/devices/internal/kubernetes/KubernetesReservationClient.kt

```diff
@@ -103,16 +113,18 @@ internal class KubernetesReservationClient(
         launch(CoroutineName("waiting-pods")) {
             @Suppress("DEPRECATION")
             podsChannel
+                .map {
+                    if (serialsChannel.isClosedForSend) {
```

Comment:
because of `distinct` operation on channels, we should check serialsChannel on the upper level